Mathlib/Analysis/Complex/BorelCaratheodory.lean

```diff
@@ -0,0 +1,414 @@
+the file defines in `divRemovableZero` a function equal to $f(z) / z$ for
+$z ≠ 0$ and equal to $f'(0)$ for $z = 0$. No restriction to $f$ with
+$f(0) = 0$ is made in this definition.
+
+In `AnalyticOn.divRemovableZero` we prove that if $f$ is analytic in an open
+set containing zero, and $f(0) = 0$ then $f(z) / z$ is analytic in the same
+open set.
+
+In `AnalyticOn.divRemovableZero_closedBall` we prove the same statement for
+functions analytic on closed balls $|z| ≤ R$. As is the proof is a little bit
+awkward, it would be more natural to prove that if $f$ is analytic in $|z| ≤ R$,
+then it is analytic on an open set containing $|z| ≤ R$ and appeal to the
+previous result.
+
+In `schwartzQuotient` we define the Schwartz quotient
+$$
+  \frac{f(z) / z}{2 * M - f(z)}
+$$
+on which the proof of Borel-Caratheodory hinges. No restrictions that
+$f(0) = 0$ or $2 M - f(z) ≠ 0$ are placed in this definition.
+
+In `AnalyticOn.schwartzQuotient` we prove that the Schwarz quotient is analytic
+in $|z| ≤ R$ provided that $f$ is analytic in $|z| ≤ R$ and $f(0) = 0$ and
+$2 * M - f(z) ≠ 0$ for all $|z| ≤ R$. The latter condition will be implied by
+the assumption $\Re{f(z)} ≤ M$ in Borel-Caratheodory.
+
+In `Complex.norm_le_norm_two_mul_sub_of_re_le` we prove an elementary inequality
+asserting that if $\Re{x} ≤ M$ then $|x| ≤ |2 M - x|$. This is used in the
+proof of Borel-Caratheodory.
+
+The function `AnalyticOn.norm_le_of_norm_le_on_sphere` is a specialization
+of the maximum modulus principle to the case of $|z| ≤ R$. Since the proof of
+Borel-Caratheodory uses the maximum modulus twice on this domain, it is handy
+to have a dedicated function for this domain. The statement allows for a
+conclusion specialized to $|z| ≤ r$ with $r ≤ R$. This is of course trivial,
+and the only reason to include such a conclusion, is because this form of
+the conclusion is used twice in the proof of Borel-Caratheodory.
+
+In `borelCaratheodory_closedBall` we prove the Borel-Caratheodory
+theorem under the assumption that $f(0) = 0$. This proof of Borel-Caratheodory
+follows the blueprint laid out in the StrongPNT project [kontorovich2025].
+
+Finally in `borelCaratheodory_closedBall'` we deduce from
+`borelCaratheodory_closedBall` a version of Borel-Caratheodory that does
+not assume vanishing of $f$ at $0$. This version matches the statement of
+the theorem as seen in Wikipedia.
+
+## References
+
+* [Alex Kontorovich et al., *PrimeNumberTheoremAnd*][kontorovich2025]
+* <https://en.wikipedia.org/wiki/Borel%E2%80%93Carath%C3%A9odory_theorem>
+
+## Tags
+
+Borel-Caratheodory, borel, caratheodory, complex, analytic, prime number theorem
+
+-/
+
+/-- Given `f`, the function `divRemovableZero f` defines a function equal
+to `f z / z` for `z ≠ 0` and equal to `deriv f 0` for `z = 0`. -/
+noncomputable abbrev divRemovableZero (f : ℂ → ℂ) : ℂ → ℂ :=
+  Function.update (fun z ↦ (f z) / z) 0 ((deriv f) 0)
```

Comment:
To me this seems overly specific. There are at least two ways that I think it can (and maybe should) be generalized:

1. I really think you should generalize away from `z = 0` and allow the replacement to occur at other points.
2. You could replace with a power of `z` in the denominator, and `iteratedDeriv`. then the lemmas would be about `MeromorphicOrderAt` instead of `f 0 = 0`.